internal/driver/driver.go

```diff
@@ -147,9 +151,54 @@ func (d *Driver) Initialize(lc logger.LoggingClient, asyncCh chan<- *sdkModels.A
 
 	// Make sure the paths of existing devices are up-to-date.
 	go d.RefreshExistingDevicePaths()
-
+	go d.StartRtspCredentialServer()
 	return nil
 }
+func (d *Driver) StartRtspCredentialServer() {
+	d.lc.Infof("starting rtsp server\n")
+	rtspAuthServer := http.NewServeMux()
+	rtspAuthServer.HandleFunc("/rtspauth", d.RtspCredentialsHandler)
+	http.ListenAndServe("localhost:8000", rtspAuthServer)
+}
+func (d *Driver) RtspCredentialsHandler(w http.ResponseWriter, r *http.Request) {
+
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		d.lc.Warnf("could not read body: %s\n", err)
+		w.WriteHeader(http.StatusUnauthorized)
+		return
+	}
+
+	var rtspRequest RtspRequest
+	err = json.Unmarshal(body, &rtspRequest)
+	if err != nil {
+		d.lc.Warnf("could not unmarshal read body: %s\n", err)
+		w.WriteHeader(http.StatusUnauthorized)
+		return
+	}
+
+	credential, edgexErr := d.tryGetCredentials(rtspAuthKey)
+	if edgexErr != nil {
+		d.lc.Warnf("failed to get credentials for at path %s", rtspRequest.Path)
```

Comment:
reword. also, we should probably add a message hinting the user that they might not have set the credentials yet:
```suggestion
		d.lc.Warnf("Failed to retrieve credentials for rtsp authentication from the secret store. Have you stored credentials yet for secretName %s?", rtspAuthKey)
```